ui/src/app/pages/homepage/data-access-requirements.spec.tsx

```diff
@@ -37,17 +38,129 @@ describe('DataAccessRequirements', () => {
     });
 
-    it('should not render the RAS module when its feature flag is disabled', () => {
+    it('should not return the RAS module from getEnabledModules when its feature flag is disabled', () => {
         serverConfigStore.set({config: {...defaultServerConfig, enableRasLoginGovLinking: false}});
-        const wrapper = component();
-        expect(findModule(wrapper, AccessModule.RASLINKLOGINGOV).exists()).toBeFalsy();
+        const [navigate, ] = useNavigation();
+        const enabledModules = getEnabledModules(allModules, navigate);
+        expect(enabledModules.includes(AccessModule.RASLINKLOGINGOV)).toBeFalsy();
+    });
+
+    it('should not return the ERA module from getEnabledModules when its feature flag is disabled', () => {
+        serverConfigStore.set({config: {...defaultServerConfig, enableEraCommons: false}});
+        const [navigate, ] = useNavigation();
+        const enabledModules = getEnabledModules(allModules, navigate);
+        expect(enabledModules.includes(AccessModule.ERACOMMONS)).toBeFalsy();
      });
 
+    it('should not return the Compliance module from getEnabledModules when its feature flag is disabled', () => {
+        serverConfigStore.set({config: {...defaultServerConfig, enableComplianceTraining: false}});
+        const [navigate, ] = useNavigation();
+        const enabledModules = getEnabledModules(allModules, navigate);
+        expect(enabledModules.includes(AccessModule.COMPLIANCETRAINING)).toBeFalsy();
+    });
+
+    it('should return the first module (2FA) from getActiveModule when no modules have been completed', () => {
+        const [navigate, ] = useNavigation();
+        const enabledModules = getEnabledModules(allModules, navigate);
+        const activeModule = getActiveModule(enabledModules, profile);
+
+        expect(activeModule).toEqual(allModules[0]);
+        expect(activeModule).toEqual(enabledModules[0]);
+
+        // update this if the order changes
+        expect(activeModule).toEqual(AccessModule.TWOFACTORAUTH)
+    });
+
+    it('should return the second module (RAS) from getActiveModule when the first module (2FA) has been completed', () => {
+        const testProfile = {
+            ...profile,
+            accessModules: {
+                modules: [{moduleName: AccessModule.TWOFACTORAUTH, completionEpochMillis: 1}]
+            }
+        };
+
+        const [navigate, ] = useNavigation();
+        const enabledModules = getEnabledModules(allModules, navigate);
+        const activeModule = getActiveModule(enabledModules, testProfile);
+
+        expect(activeModule).toEqual(allModules[1]);
+        expect(activeModule).toEqual(enabledModules[1]);
+
+        // update this if the order changes
+        expect(activeModule).toEqual(AccessModule.RASLINKLOGINGOV)
+    });
+
+    // update this if the order changes
+    it('should return the second enabled module (ERA, not RAS) from getActiveModule' +
```

Comment:
a bit of paranoia here ... but it shows that this module cannot become active when disabled